rust/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -127,34 +128,38 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
         let (ism_addresses, threshold) = ism.modules_and_threshold(message).await.context(CTX)?;
         let threshold = threshold as usize;
 
-        let metas = join_all(
-            ism_addresses
-                .iter()
-                .map(|ism_address| self.base.build(*ism_address, message)),
-        )
-        .await;
-
-        let sub_modules = join_all(
+        let sub_modules_and_metas = join_all(
             ism_addresses
                 .iter()
-                .map(|ism_address| self.base.build_ism(*ism_address)),
+                .map(|ism_address| self.base.build_ism_and_metadata(*ism_address, message)),
         )
         .await;
 
-        let filtered_sub_module_metas = metas
+        // Partitions things into
+        // 1. ok_sub_modules: ISMs with metadata with valid metadata
+        // 2. err_sub_modules: ISMs with invalid metadata
+        let no_err_sub_modules_and_metas: Vec<_> = sub_modules_and_metas
             .into_iter()
-            .enumerate()
-            .zip(sub_modules.into_iter())
-            .filter_map(|((index, meta_result), sub_module_result)| {
-                match (meta_result, sub_module_result) {
-                    (Ok(Some(meta)), Ok(ism)) => Some(IsmAndMetadata::new(ism, index, meta)),
-                    _ => None,
-                }
-            })
+            .filter_map(|sub_module_and_meta| sub_module_and_meta.ok())
             .collect();
-
+        let (ok_sub_modules, err_sub_modules): (Vec<_>, Vec<_>) = no_err_sub_modules_and_metas
```

Comment:
So my I was thinking was:
1. we want the union of the partitions `ok_sub_modules U err_sub_modules` to give us all the submodules in the aggregation ISM
2. for any err_sub_modules, we want to have as much info as possible about these sub_modules so we can log it. The most useful info is the ISM address and the module type, if we have it
  a. An error could occur for any of the reasons [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/agents/relayer/src/msg/metadata/base.rs#L77-L103), which includes being unable to build an ISM (likely an RPC error), unable to get the module_type (ism.module_type() could revert, or RPC error), too deep of a clone of base, invalid ModuleType, or building the metadata failed

Because of (1) I was imagining we wouldn't have the filter_map up above at all, and instead when we do the partition here we'd iter through the `sub_modules_and_meta` directly, and if one was an Err then we wouldn't have access to the module type, which is why I suggested it be optional. Atm I think `err_sub_modules` will just include submodules where we were just unable to build the metadata, but all other fallible steps were successful. Which probably covers most of the cases, but I think it'd be good to cover all the cases so we get as much info as possible when we log `err_isms`

So the cases for `err_sub_modules` would be:
- Result::Err
- Result::Ok, but metadata was None

In which case I wonder if we should just have err_sub_modules / err_isms be typed as something like `Vec<(H256, Result<IsmWithMetadataAndType>)>` so we get the addresses & result for each of the sub modules with errors